src/Adapter/MSTest.TestAdapter/MSTestSettings.cs

```diff
@@ -831,4 +850,162 @@ private static void SetGlobalSettings(
+        ParseBooleanSetting(configuration, "treatDiscoveryWarningsAsErrors", logger, result => settings.TreatDiscoveryWarningsAsErrors = result);
+        ParseBooleanSetting(configuration, "considerEmptyDataSourceAsInconclusive", logger, result => settings.ConsiderEmptyDataSourceAsInconclusive = result);
+        ParseBooleanSetting(configuration, "treatClassAndAssemblyCleanupWarningsAsErrors", logger, result => settings.TreatClassAndAssemblyCleanupWarningsAsErrors = result);
+        ParseBooleanSetting(configuration, "considerFixturesAsSpecialTests", logger, result => settings.ConsiderFixturesAsSpecialTests = result);
+        ParseBooleanSetting(configuration, "cooperativeCancellationTimeout", logger, result => settings.CooperativeCancellationTimeout = result);
+        ParseBooleanSetting(configuration, "orderTestsByNameInClass", logger, result => settings.OrderTestsByNameInClass = result);
+        ParseBooleanSetting(configuration, "parallelism:enabled", logger, result => settings.OrderTestsByNameInClass = result);
+
+        ParseIntegerSetting(configuration, "timeout:test", logger, result => settings.TestTimeout = result);
+        ParseIntegerSetting(configuration, "timeout:assemblyCleanup", logger, result => settings.AssemblyCleanupTimeout = result);
+        ParseIntegerSetting(configuration, "timeout:assemblyInitialize", logger, result => settings.AssemblyInitializeTimeout = result);
+        ParseIntegerSetting(configuration, "timeout:classInitialize", logger, result => settings.ClassInitializeTimeout = result);
+        ParseIntegerSetting(configuration, "timeout:classCleanup", logger, result => settings.ClassCleanupTimeout = result);
+        ParseIntegerSetting(configuration, "timeout:testInitialize", logger, result => settings.TestInitializeTimeout = result);
+        ParseIntegerSetting(configuration, "timeout:testCleanup", logger, result => settings.TestCleanupTimeout = result);
+
+        ParseStringSetting(configuration, "settingsFile", logger, fileName => settings.TestSettingsFile = fileName);
+
+        if (configuration["mstest:classCleanupLifecycle"] is string classCleanupLifecycle)
+        {
+            if (TryParseEnum(classCleanupLifecycle, out ClassCleanupBehavior lifecycle))
+            {
+                settings.ClassCleanupLifecycle = lifecycle;
+            }
+            else
+            {
+                throw new AdapterSettingsException(string.Format(
+                    CultureInfo.CurrentCulture,
+                    Resource.InvalidClassCleanupLifecycleValue,
+                    classCleanupLifecycle,
+#if NET
+                    string.Join(", ", Enum.GetNames<ClassCleanupBehavior>())));
+#else
+                    string.Join(", ", EnumPolyfill.GetNames<ClassCleanupBehavior>())));
+#endif
+            }
+        }
+
+        if (configuration["mstest:parallelism:workers"] is string workers)
+        {
+            settings.ParallelizationWorkers = int.TryParse(workers, out int parallelWorkers)
```

Comment:
wonder if we should warning a message in case the value is specified but actually is not a good int to parse.
Fine go on with the default proc count, but looks an important settings that users want to be correct and it's impacting the runtime.

Reply:
yup we are already throwing in case of invalid value